backend/src/routes/pm/pm.gateway.ts

```diff
@@ -40,31 +47,66 @@ export class PmGateway
     );
     this.server.use(socketAuth(this.sessionService));
     this.server.use(addUser);
+    this.server.use(addBlock);
   }
 
   async handleConnection(@ConnectedSocket() client: Socket) {
-    const handshake = client.handshake as UserHandshake;
-
+    const handshake = client.handshake as BlockHandshake;
     client.join(handshake.user.id.toString());
 
     handshake.user.status = 1;
     await handshake.user.save();
   }
 
   async handleDisconnect(@ConnectedSocket() client: Socket) {
-    const handshake = client.handshake as UserHandshake;
+    const handshake = client.handshake as BlockHandshake;
 
     handshake.user.status = 0;
     await handshake.user.save();
   }
+  @SubscribeMessage("ban")
+  async handleban(
+    @ConnectedSocket() client: Socket,
+    @MessageBody("id") id: number
+  ) {
+    const handshake = client.handshake as BlockHandshake;
+    if (isNil(id) || isNaN(id)) return;
+    if (handshake.user.id === id) return;
+    if (handshake.block.has(id)) return;
+    await this.blockService.blockUser(handshake.user.id, id);
+    //todo: update blocked_by if user blocked is connected
+    const sockets = await this.server.fetchSockets();
+    const userSocket = find(pathEq(["handshake", "user", "id"], id))(sockets);
 
+    if (!userSocket) return;
+    userSocket.handshake.block.add(handshake.user.id);
+    userSocket.handshake.user.blocked_by = await this.blockService.getblocker(
```

Comment:
This line is not needed, we only use handshake.block, handshake.user.blocked_by does not exist, and can lead to crash when using user.save() to update status.

Reply:
agreed, it is now remvoed